### DISCUSSION TITLE
Can the open source version get browserVersion and osVersion?

### DISCUSSION THREAD

**Discussion by AlbertXiaoPeng** [fingerprintjs/fingerprintjs#975] [Q&A] [answered]
I want to get osVersion, os, browserName and browserVersion in the open source version.

### When I use it as follows

```html
<script>
    // Initialize the agent at application startup.
    const fpPromise = import('https://openfpcdn.io/fingerprintjs/v4')
        .then(FingerprintJS => FingerprintJS.load())

    // Get the visitor identifier when you need it.
    fpPromise
        .then(fp => fp.get({extendedResult: true}))
        .then(result => {
            console.log(result)
        })
</script>
```

### The results are as follows

![image](https://github.com/fingerprintjs/fingerprintjs/assets/44151574/3c29ad86-f40b-413c-9277-72057797c981)

It doesn't have the values I need.

**Comment by AlbertXiaoPeng** [comment]
Are these features only accessible to the pro version?

**Comment by Finesse** [comment, marked as answer]
Getting browser and OS version is outside of the FingerprintJS scope. You can use [UAParser.js](https://uaparser.js.org) for this.

**Reply by AlbertXiaoPeng** [reply]
We used UAParser.js, but it was not recognized correctly in Windows 11, so I wanted to try to get it right when I saw that the [fingerprint](https://dev.fingerprint.com/docs/js-agent#os) documentation existed for getting the OS
